modules/core/src/main/scala/com.snowplowanalytics.iglu/client/validator/CirceValidator.scala

```diff
@@ -88,25 +88,30 @@ object CirceValidator extends Validator[Json] {
     V4SchemaInstance.getSchema(new ObjectMapper().readTree(MetaSchemas.JsonSchemaV4Text))
 
   def validate(data: Json, schema: Json): Either[ValidatorError, Unit] = {
-    val jacksonJson = circeToJackson(schema)
+    val jacksonJson = circeToJackson(schema, Int.MaxValue)
     evaluateSchema(jacksonJson)
       .flatMap { schema =>
-        validateOnReadySchema(schema, data).leftMap(ValidatorError.InvalidData.apply)
+        validateOnReadySchema(schema, data, Int.MaxValue).leftMap(ValidatorError.InvalidData.apply)
       }
   }
 
-  def checkSchema(schema: Json): List[ValidatorError.SchemaIssue] = {
-    val jacksonJson = circeToJackson(schema)
+  @deprecated("Use `checkSchema(schema, maxJsonDepth)`", "3.2.0")
+  def checkSchema(schema: Json): List[ValidatorError.SchemaIssue] =
+    checkSchema(schema, Int.MaxValue)
```

Comment:
This method is used by:

- `Client` (via `Validator`) which I think we can deprecate
- igluctl, which could switchover to use `checkSchema(schema, maxJsonDepth)`